bin/sozo/src/commands/execute.rs

```diff
@@ -1,55 +1,102 @@
-
-    #[arg(short, long)]
-    #[arg(help = "The calldata to be passed to the system. Comma separated values e.g., \
-                  0x12345,128,u256:9999999999. Sozo supports some prefixes that you can use to \
-                  automatically parse some types. The supported prefixes are:
-                  - u256: A 256-bit unsigned integer.
-                  - sstr: A cairo short string.
-                  - str: A cairo string (ByteArray).
-                  - int: A signed integer.
-                  - no prefix: A cairo felt or any type that fit into one felt.")]
-    pub calldata: Option<String>,
+    pub calls: Vec<String>, 
+
+    #[arg(long)]
+    #[arg(help = "If true, sozo will compute the diff of the world from the chain to translate \
+                  tags to addresses.")]
+    pub diff: bool, 
 
     #[command(flatten)]
-    pub starknet: StarknetOptions,
+    pub starknet: StarknetOptions, 
 
     #[command(flatten)]
     pub account: AccountOptions,
 
     #[command(flatten)]
-    pub world: WorldOptions,
+    pub world: WorldOptions, 
 
     #[command(flatten)]
-    pub transaction: TransactionOptions,
+    pub transaction: TransactionOptions, 
+}
+
+#[derive(Debug)]
+pub struct CallArgs {
+    pub tag_or_address: ResourceDescriptor, // Contract address or tag
+    pub entrypoint: String, // Entrypoint to call
+    pub calldata: Option<String>, // Calldata to pass to the entrypoint
+}
+
+
+impl std::str::FromStr for CallArgs {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let s = s.trim();
+        if s.is_empty() {
+            return Err(anyhow!("Empty call string"));
+        }
+
+        let parts: Vec<&str> = s.split(',').collect();
+        if parts.len() < 2 {
+            return Err(anyhow!("Invalid call format. Expected format: <CONTRACT_NAME>,<ENTRYPOINT_NAME>,<ARG1>,<ARG2>,..."));
+        }
+
+        let entrypoint = parts[1].trim();
+        if entrypoint.is_empty() {
+            return Err(anyhow!("Empty entrypoint"));
+        }
+        if !entrypoint.chars().all(|c| c.is_ascii_alphanumeric() || c == '_') {
+            return Err(anyhow!("Invalid entrypoint format. Must contain only alphanumeric characters and underscores"));
+        }
+
+        Ok(CallArgs {
+            tag_or_address: parts[0].parse()?,
+            entrypoint: entrypoint.to_string(),
+            calldata: if parts.len() > 2 { Some(parts[2..].join(",")) } else { None },
+        })
+    }
+}
+
+fn resolve_contract_address(
+    descriptor: &ResourceDescriptor,
+    world_diff: &WorldDiff,
+) -> Result<Address> {
+    match descriptor {
+        ResourceDescriptor::Address(address) => Ok(*address),
+        ResourceDescriptor::Tag(tag) => {
+            let selector = naming::compute_selector_from_tag(tag);
+            world_diff
+                .get_contract_address(selector)
+                .ok_or_else(|| anyhow!("Contract {descriptor} not found in the world diff."))
```

Comment:
This is not the way you can get the contract address, since the world diff is not always computed if we have a local manifest.

https://github.com/dojoengine/dojo/blob/81e9707c964bf74b2a4d34831327feedd7612a18/bin/sozo/src/commands/execute.rs#L81